presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -113,70 +121,124 @@ public void testOffsetPushDown() throws ExecutionException, InterruptedException
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 998);
+        String dumpMsg = buildDebugDumpString(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpMsg)
+                .isEqualTo(998);
 
         // timestamp_upper_bound_force_push_down_enabled set as true.
         Session sessionWithUpperBoundPushDownEnabled = Session.builder(getSession())
                 .setSystemProperty("kafka.timestamp_upper_bound_force_push_down_enabled", "true")
                 .build();
 
         queryResult = queryRunner.executeWithQueryId(sessionWithUpperBoundPushDownEnabled, sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+        dumpMsg = buildDebugDumpString(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpMsg)
+                .isEqualTo(2);
     }
 
-    @Test
+    @Test(retryAnalyzer = FixedCountRetryAnalyzer.class)
     public void testTimestampLogAppendModePushDown() throws ExecutionException, InterruptedException
     {
-        Pair<String, String> timePair = createTimestampTestMessages(topicNameLogAppend);
+        RecordMessage recordMessage = createTimestampTestMessages(topicNameLogAppend);
         DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
         // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
-        String sql = String.format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameLogAppend, timePair.first(), timePair.second());
+        String sql = format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
+                topicNameLogAppend, recordMessage.getStartTime(), recordMessage.getEndTime());
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+
+        String dumpMsg = buildDebugDumpString(topicNameLogAppend, recordMessage, queryRunner, sql, queryResult);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpMsg)
+                .isEqualTo(2);
+    }
+
+    private String buildDebugDumpString(String topic, RecordMessage recordMessage, DistributedQueryRunner queryRunner, String sql, ResultWithQueryId<MaterializedResult> queryResult)
```

Comment:
nit: this signature seems long enough to justify putting each argument in separate line